test/dub_package/source/app.d

```diff
@@ -43,3 +43,119 @@ unittest
     scope parser = new Parser!ASTBase(null, null, false);
     assert(parser !is null);
 }
+
+
+// inits the DMD compiler
+void init()
+{
+    import ddmd.globals : global;
+
+    global._init;
+    global.params.isLinux = 1;
+    import ddmd.dmodule : Module;
+    import ddmd.id : Id;
+    import ddmd.mtype : Type;
+    import ddmd.target : Target;
+    import ddmd.expression : Expression;
+    import ddmd.objc : Objc;
+    import ddmd.builtin : builtin_init;
+
+    // Initialization
+    Type._init();
+    Id.initialize();
+    Module._init();
+    Target._init();
+    Expression._init();
+    Objc._init();
+    builtin_init();
+}
+
+// add import paths
+void addImports(string[] path)
+{
+    import ddmd.globals : global;
+    import ddmd.arraytypes : Strings;
+
+    Strings* res = new Strings();
+    foreach (p; path)
+    {
+        import std.string : toStringz;
+        Strings* a = new Strings();
+        a.push(p.toStringz);
+        res.append(a);
+    }
+    global.path = res;
+}
+
+// frontend
+unittest
+{
+    import ddmd.astcodegen : ASTCodegen;
+    import ddmd.dmodule : Module;
+    import ddmd.globals : Loc;
+    import ddmd.parse : Parser;
+    import ddmd.statement : Identifier;
+    import ddmd.tokens : TOKeof;
+    import ddmd.id : Id;
+
+    init;
```

Comment:
We still need to define all predefined versions, but `addDefaultVersionIdentifiers` is `private`.
Imho if I would be about to write an IDE plugin, I wouldn't want to keep this list in sync if DMD already has the logic.
For example, to import Phobos on Linux, I need this list:

```
import ddmd.cond : VersionCondition;
VersionCondition.addPredefinedGlobalIdent("DigitalMars");
VersionCondition.addPredefinedGlobalIdent("Posix");
VersionCondition.addPredefinedGlobalIdent("linux");
VersionCondition.addPredefinedGlobalIdent("ELFv1");
VersionCondition.addPredefinedGlobalIdent("LittleEndian");
VersionCondition.addPredefinedGlobalIdent("CRuntime_Glibc");
VersionCondition.addPredefinedGlobalIdent("D_Version2");
VersionCondition.addPredefinedGlobalIdent("all");
VersionCondition.addPredefinedGlobalIdent("D_InlineAsm_X86_64");
VersionCondition.addPredefinedGlobalIdent("X86_64");
VersionCondition.addPredefinedGlobalIdent("D_HardFloat");
```

So any chance we can refactor `addDefaultVersionIdentifiers` and expose it?